pandas/core/frame.py

```diff
@@ -3924,12 +3925,14 @@ def isetitem(self, loc, value) -> None:
                 )
 
             for i, idx in enumerate(loc):
-                arraylike, _ = self._sanitize_column(value.iloc[:, i])
-                self._iset_item_mgr(idx, arraylike, inplace=False)
+                arraylike, refs = self._sanitize_column(
+                    value.iloc[:, i], using_cow=using_cow
```

Comment:
Could we actually call `using_copy_on_write()` inside `_sanitize_column` instead of having to pass it everywhere?

Reply:
Yes this will be a clean up when all the other PRs are merged